internal/rbd/controllerserver.go

```diff
@@ -96,7 +96,7 @@ func (cs *ControllerServer) validateVolumeReq(ctx context.Context, req *csi.Crea
 
 func (cs *ControllerServer) parseVolCreateRequest(
 	ctx context.Context,
-	req *csi.CreateVolumeRequest) (*rbdVolume, error) {
+	req *csi.CreateVolumeRequest, secrets map[string]string) (*rbdVolume, error) {
```

Comment:
This is quite ugly, the `CreateVolumeRequest` already contains secrets. Preventing to pass extra (or replacement) `secrets` would be much better. If there is really no other reasonable way to do this, it needs to be well documented for the function.

Reply:
@nixpanic sure, let me check , considering the secret parsing and using done in many places ( I mean here and there, ie sometimes from helper directly on request, sometimes direct usage of passed cred ..etc) I noticed some difficulty.. but let me recheck again.

Reply:
> If there is really no other reasonable way to do this, it needs to be well documented for the function.
as we decided to not change the request itself,  this is the reasonable way to do this. I have also documented parseVolCreateRequest() with the details.. thanks.